pkg/virt-launcher/virtwrap/api/defaults.go

```diff
@@ -77,7 +77,9 @@ func (d *Defaulter) setDefaults_Features(spec *DomainSpec) {
 		spec.Features = &Features{}
 	}
 
-	spec.Features.VMPort = &FeatureState{State: "off"}
+	if spec.Features.VMPort != nil {
+		spec.Features.VMPort = &FeatureState{State: "off"}
+	}
```

Comment:
I think we can simply do:
```go
	if !d.isARM64() {
		spec.Features.VMPort = &FeatureState{State: "off"}
	}
```

I find it more readable and less confusing. WDYT?

Reply:
Yeah, makes sense.
My problem is that when reading this code it is unclear on which circumstances `spec.Features.VMPort` would not be nil.

I think it's better to delete this code from here (i.e. not default it to anything) and instead deal with all cases in the same place under `Convert_v1_VirtualMachineInstance_To_api_Domain()`. Does that make sense?

I'm suggesting something like:
```go
		if c.Architecture.isVMPortNeeded() {
			domain.Spec.Features.VMPort = &api.FeatureState{}
		} else {
			spec.Features.VMPort = &FeatureState{State: "off"}
		}
```

Reply:
I'm fine with it. Let's see if @xpivarc has any suggestions; otherwise, I'll modify the code.

Reply:
We need to add a converter unit test to prevent this